pkg/utils/utils.go

```diff
@@ -31,6 +32,25 @@ func MergeByteMap(dst, src map[string][]byte) map[string][]byte {
 	return dst
 }
 
+// ConvertName converts a string into a secret-key compatible string.
+// Replaces any non-alphanumeric characters with its unicode code.
+func ConvertName(in string) string {
```

Comment:
I made a comment regarding the mapping here: https://github.com/external-secrets/external-secrets/issues/698#issuecomment-1062133582

Reply:
Applied new strategy. I moved this out of the scope of each provider as well, and done as a general check (as I believe it made more sense)